### ISSUE TITLE
Time duration sliders not working

### ISSUE THREAD

**Issue by thanksegon** [StephenBlackWasAlreadyTaken/NightWatch#111]
Using latest apk I am unable to select a snooze time other than default. I click other times/choices but they are not saved or recognized. For example, the default snooze for a High is 2 hours; I'd rather set it to 1 hour. I choose 60 minutes in the slider but it is not accepted. Only Cancel is recognized. The snooze stays set to the default of 2 hours.

**Comment by GHNS** [comment]
I can confirm this. It is because the "Set" button is not visible (more precisely "not being shown"). 

I downloaded the "Studio 3" branch and played around with the layout_width-values in mobile/src/main/res/layout/snooze_picker.xml

These are the values that made the set buttons appear:

```
<?xml version="1.0" encoding="utf-8"?>
<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
    android:orientation="vertical" android:layout_width="fill_parent"
    android:layout_height="fill_parent">

    <LinearLayout
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_gravity="center"
        android:orientation="vertical"
        android:gravity="center_horizontal">

        <NumberPicker
                android:id="@+id/numberPicker1"
                android:layout_width="fill_parent"
                android:layout_height="fill_parent"
                android:layout_alignParentTop="true"
                android:layout_centerHorizontal="true"/>

        <LinearLayout
            android:orientation="horizontal"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_gravity="center_horizontal"
            android:gravity="center_horizontal">

            <Button
                    android:id="@+id/button2"
                    android:layout_width="wrap_content"
                    android:layout_height="wrap_content"
                    android:text="Cancel" />

            <Button
                    android:id="@+id/button1"
                    android:layout_width="wrap_content"
                    android:layout_height="wrap_content"
                    android:text="Set" />
        </LinearLayout>
    </LinearLayout>

</LinearLayout>
```
![screenshot](https://user-images.githubusercontent.com/40469911/42179060-ed36d7fa-7e32-11e8-9478-88e7dd923f7b.png)

**Comment by GHNS** [comment]
Do you mean the snooze-timespan which is shown below the snooze-button or the dialog that pops-up when you tap on "disable XXX alterts"?

I think for the first you have to select the time being used when you press the "snooze" button (which seems to work, if I select 10 minutes as snooze time and click on snooze, it tells me the current alert is snoozed until now+10mins).

For the latter (dialog for choosing time to disable alerts) there is an issue that prevents the "set" button from being shown. I have fixed the issue on my device by recompiling the project with a changed /mobile/src/main/res/layout/snooze_picker.xml file (I changed layout_width and layout_height values):

```
<?xml version="1.0" encoding="utf-8"?>
<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
    android:orientation="vertical" android:layout_width="fill_parent"
    android:layout_height="fill_parent">

    <LinearLayout
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_gravity="center"
        android:orientation="vertical"
        android:gravity="center_horizontal">

        <NumberPicker
                android:id="@+id/numberPicker1"
                android:layout_width="fill_parent"
                android:layout_height="fill_parent"
                android:layout_alignParentTop="true"
                android:layout_centerHorizontal="true"/>

        <LinearLayout
            android:orientation="horizontal"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_gravity="center_horizontal"
            android:gravity="center_horizontal">

            <Button
                    android:id="@+id/button2"
                    android:layout_width="wrap_content"
                    android:layout_height="wrap_content"
                    android:text="Cancel" />

            <Button
                    android:id="@+id/button1"
                    android:layout_width="wrap_content"
                    android:layout_height="wrap_content"
                    android:text="Set" />
        </LinearLayout>
    </LinearLayout>

</LinearLayout>
```
![screenshot](https://user-images.githubusercontent.com/40469911/42179520-96536c58-7e34-11e8-8c14-34174e668aed.png)

**Comment by thanksegon** [comment]
Hi, I've attached a screenshot. I want to change this to 15 minutes, for example. I click on the 15 minutes and nothing happens. There isn't a readily apparent way to select 15 minutes and the save the alert.


Joe Ritson
Simpletonia LLC
Office Phone: 617.595.4900

Cell Phone: 617.775.7750

LinkedIn: www.linkedin.com/in/josephritson<http://www.linkedin.com/in/josephritson>



________________________________

**Comment by thanksegon** [comment]
Just curious if NW is an active project? Will there be updates and fixes, or have resources moved to another project?

**Comment by Wanitta** [comment]
I used Nightwatch for awhile. With the newer versions of xDrip+, I found I didn't need it. Xdrip+ App will connect directly to various Android wear watches and can be set up with a follower. Check out xDrip group on Facebook.

Wanitta


Sent from my Verizon, Samsung Galaxy smartphone


-------- Original message --------

**Comment by thanksegon** [comment]
Thanks. I’m using xDrip-jswhich runs on my rig, so there isn’t an app per se.  Watch is the only Android app someone can use to follow me, I think.

Joe Ritson
Cell Phone: 617.775.7750
LinkedIn: www.linkedin.com/in/josephritson<http://www.linkedin.com/in/josephritson>

**Comment by Wanitta** [comment]
You can put xDrip+ app on both phones and set one up as Master and the other as Follower. I gave you the info on where to look for more info.

Wanitta



Sent from my Verizon, Samsung Galaxy smartphone


-------- Original message --------

**Comment by thanksegon** [comment]
GHNS I'm not a developer, but it sounds as I should download the source, modify the snooze_picker.xlm and then compile a new apk? Is that what you're suggesting I do?

**Comment by GHNS** [comment]
> 
> 
> GHNS I'm not a developer, but it sounds as I should download the source, modify the snooze_picker.xlm and then compile a new apk? Is that what you're suggesting I do?

Yes, that's true. I downloaded the studio3 branch so I could open it in Android Studio 3. Then, I edited the snooze_picker.xlm file from within Android Studio (you can see the effect of changes on the preview panel immediately) and built a signed APK. You need to create a new Key store for doing so. Once you have built the APK, you need to uninstall NightWatch from your mobile (write your app settings down before) and then install your own APK (because you cannot in-place update the app using your own key).

Good luck!
